apps/meteor/client/views/room/hooks/useOpenRoomMutation.tsx

```diff
@@ -0,0 +1,45 @@
+import type { RoomType } from '@rocket.chat/core-typings';
+import { useEndpoint, useUserId } from '@rocket.chat/ui-contexts';
+import { useMutation } from '@tanstack/react-query';
+
+import { updateSubscription } from '../../../lib/mutationEffects/updateSubscription';
+
+const openEndpoints = {
+	p: '/v1/groups.open',
+	d: '/v1/im.open',
+	c: '/v1/channels.open',
+	v: '/v1/channels.open',
+	l: '/v1/channels.open',
+} as const;
+
+export const useOpenRoomMutation = ({ type }: { type: RoomType }) => {
+	const openRoom = useEndpoint('POST', openEndpoints[type]);
+	const userId = useUserId();
```

Comment:
If the mutation depends so hardly of `userId` being not-null, consider short-cutting the hook here with an `throw` statement:

```suggestion
	const userId = useUserId();
	
	if (!userId) {
		throw new Error('error-invalid-user');
	}
```

There is no reason to give component's access to a mutation that is going to fail regardless.